### ISSUE TITLE
"TypeError: Cannot read property 'clone' of undefined" after upgrading to 1.5.0

### ISSUE THREAD

**Issue by doberkofler** [tracker1/gulp-header#21]
After upgrading to 1.5.0, I get the following error stack when trying to run gulp-header.
Any help would be appreciated.

```
D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\index.js:58
    var file = _file.clone({ contents: false });
                    ^
TypeError: Cannot read property 'clone' of undefined
    at DestroyableTransform.EndStream [as _flush] (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\index.js:58:21)
    at DestroyableTransform.<anonymous> (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\node_modules\through2\node_modules\readable-stream\lib\_stream_transform.js:123:12)
    at DestroyableTransform.g (events.js:199:16)
    at DestroyableTransform.emit (events.js:104:17)
    at prefinish (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\node_modules\through2\node_modules\readable-stream\lib\_stream_writable.js:492:12)
    at finishMaybe (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\node_modules\through2\node_modules\readable-stream\lib\_stream_writable.js:500:7)
    at endWritable (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\node_modules\through2\node_modules\readable-stream\lib\_stream_writable.js:512:3)
    at DestroyableTransform.Writable.end (D:\MyDev\ljs_app\trunk\periscope\build\node_modules\gulp-header\node_modules\through2\node_modules\readable-stream\lib\_stream_writable.js:477:5)
    at Stream.onend (stream.js:79:10)
    at Stream.emit (events.js:129:20)
```

**Comment by mjhamm75** [comment]
+1
Broken in PROD with this error right now.

**Comment by doberkofler** [comment]
I was able to gather some more information and have found a simple workaround:
* The problems seem to happen when gulp.src() does not find any files.
* The problem can be worked around by moving all the chained gulp tasks in an "if block" that first checks if there are any files to process.

**Comment by OndeVai** [comment]
+1 -- this is hosing us too :)

**Comment by mpalermo** [comment]
+1 -- Can't build right now because of this :)

**Comment by mjhamm75** [comment]
+1 on the workaround.  Thanks doberkofler.

**Comment by OndeVai** [comment]
SInce this plugin is being used in another plugin, gulp-angular-templatecache, there's no workaround for us...

**Comment by adgoncal** [comment]
This is the same error I reported on #17 and it seems there is already a solution in a pull request #19.

**Comment by kobezzza** [comment]
+1

**Comment by DOrlov77** [comment]
+1

**Comment by mohsen1** [comment]
Stuff like this make me question if having flexible dependency versions is a right thing to do at all...

**Comment by remento** [comment]
For gulp-angular-templatecache users, you can seed the npm cache with an older version gulp header

    npm cache clean
    npm uninstall gulp-angular-templatecache
    npm install gulp-header@1.2.2 --save-dev
    npm install gulp-angular-templatecache --save-dev

To see what version of gulp header you are using…

    npm list gulp-header

Note that the version of gulp-header to avoid is version 1.5

**Comment by tracker1** [comment]
this should be resolved in `1.5.1` apologies for this... I probably should have just done a full version bump, but didn't as the interface didn't actually change, but sourcemaps was added and figured minor was more appropriate.
